graph/src/lib/clientGraph.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Starter LangGraph.js Template
+ * Make this code your own!
+ */
+import { ToolNode } from '@langchain/langgraph/prebuilt';
+import { ChatOpenAI } from '@langchain/openai';
+import { tokensSearch, bebopRate, EvmKit } from '@agentic-chat/tools';
+import { SYSTEM_PROMPT } from '@agentic-chat/utils';
+import {
+  END,
+  MemorySaver,
+  MessagesAnnotation,
+  START,
+  StateGraph,
+} from '@langchain/langgraph/web';
+import { WalletClient } from 'viem';
+import { AIMessage, SystemMessage } from '@langchain/core/messages';
+import { RunnableLambda } from '@langchain/core/runnables';
+
+// @ts-expect-error TODO: FIXME maybe
+const env = import.meta?.env ? import.meta.env : process.env;
+
+const model = new ChatOpenAI({
+  modelName: 'gpt-4o-mini',
+  temperature: 0,
+  openAIApiKey: env.VITE_OPENAI_API_KEY,
+});
+
+// Adds persistence
+const checkpointer = new MemorySaver();
+
+function shouldContinue(
```

Comment:
I'd expect this to return a boolean based on the name. Is there a better name we can use?

Reply:
Not sure about this one - this is the official terminology used in examples (see https://langchain-ai.github.io/langgraphjs/tutorials/quickstart/) and once you get used to the flow, it *does* actually make sense as it directs the graph to continue or not, I personally find this way clearer than something like `conditionalToolsOrEndEdge`, WDYT @0xApotheosis?

Reply:
Yeah that's fair, let's stick to their convention 🙏